wallet-new/src/Cardano/Wallet/WalletLayer/Legacy.hs

```diff
@@ -128,21 +150,21 @@ pwlGetWalletIds = do
     ws          <- askWalletSnapshot
     migrate $ getWalletAddresses ws
 
+
 pwlGetWallet
     :: forall ctx m. (MonadLegacyWallet ctx m)
     => WalletId
-    -> m (Maybe Wallet)
+    -> m Wallet
 pwlGetWallet wId = do
     ws          <- askWalletSnapshot
 
     cWId        <- migrate wId
     wallet      <- V0.getWallet cWId
 
-    pure $ do
+    maybeThrow (WalletNotFound wId) $ do
```

Comment:
I would prefer to see this written as:

```haskell
pwlGetWallet :: WalletId -> m (Either GetWalletError Wallet)
```

(provided that `GetWalletError` contains more information and is more valuable than `Nothing`)

We should really be trying to avoid `MonadThrow` as much as possible and representing things as values whenever possible. `MonadThrow` is a way of evading the type system, and we have no need to do that in simple cases like `pwlGetWallet`.